lib/Dancer2/Core/Route.pm

```diff
@@ -9,6 +9,14 @@ use Scalar::Util 'blessed';
 
 our ( $REQUEST, $RESPONSE, $RESPONDER, $WRITER, $ERROR_HANDLER );
 
+my $count = 0;
+
+has name => (
+    is      => 'ro',
+    isa     => Str,
```

Comment:
I suggest requiring route names to be truthy for consistency with my next comment.

Reply:
This can be verified when you create a new one. We will test when a named route is created.

Reply:
`Types::Standard` doesn't have `True()` as a type. There is `Defined()` which I believe all strings are anyway, as well as `0` as a value.

Is there are particular type value you're suggesting or just replacing it with `sub { defined && length && !!$_ }` or something?

Reply:
@xsawyerx That `sub` indeed covers what I was thinking of.  I think `sub { !!$_ }` would be sufficient:
- `!!$_` true implies `defined`.  
- As for `length`, I'd say a user who goes to the trouble of making an empty string truthy should not expect any help with issues :laughing:

Reply:
I eventually replaced it with `has_name()` predicate because I needed to know when a route name can be used or not.

Providing a default name might conflict with a name a user provides. Imagine we give them numbers and a user does the same - there's a conflict. I'm not sure we should be automatically naming them. This can also lead to a possible accidentally call to `uri_for_route()` in which the name is typo'ed but since we named it, it hits a different route.

Reply:
Resolved